DemoApp/DemoAppUtils.swift

```diff
@@ -14,3 +17,25 @@ extension URL {
         }
     }
 }
+
+extension View {
+
+    func observeAndHandleDeeplinks(
+        _ viewModel: CallViewModel,
+        deeplinkInfoPublisher: AnyPublisher<DeeplinkInfo, Never>,
+        resetAppState: @escaping () -> Void
+    ) -> some View {
+        self.onReceive(deeplinkInfoPublisher) { deeplinkInfo in
+            if deeplinkInfo != .empty {
+                /// https://github.com/apple/swift/pull/60688
+                Task { @MainActor in
+                    viewModel.joinCall(
```

Comment:
I think it would be more complicated than this:
- What happens if we are not logged in? 
- What happens if we are still not connected to the web socket?
- What happens if the token is expired?

We will also probably support a case where the token is provided in the URL, or we join anonymously or as guest.

Reply:
> What happens if we are not logged in?
I see that `HomeView` is only available to `loggedIn` users. In the case you are not `loggedIn` the view won't be on the hierarchy (or even created) so there shouldn't be any issue there.

> What happens if we are still not connected to the web socket?
> What happens if the token is expired?

Don't we already handle those scenarios in the `joinCall` as prerequisites? 

> We will also probably support a case where the token is provided in the URL, or we join anonymously or as guest.
I'm not sure how (probably i don't have enough context) this is going to affect it